cpp/inc/bond/core/value.h

```diff
@@ -209,8 +209,7 @@ class value_common
 
     // skip value of non-matching type
     template <typename Protocols = BuiltInProtocols, typename X>
-    typename boost::disable_if_c<is_matching<T, X>::value>::type
-    Deserialize(X& /*var*/) const
+    void Deserialize(X& /*var*/, typename boost::disable_if<is_matching<T, X> >::type* = nullptr) const
```

Comment:
This changes the function signature so that the similar overloads in derived types won't hide this one.